### DISCUSSION TITLE
Is there a SpanExporter that supports OTLP/HTTP?

### DISCUSSION THREAD

**Discussion by wdengw** [open-telemetry/opentelemetry-java#3280] [Q&A]
Looking for a SpanExporter that can send trace using OTLP/HTTP.

**Comment by kenfinnigan** [comment]
https://github.com/open-telemetry/opentelemetry-java/tree/main/exporters/otlp/trace supports OTLP but with gRPC.

I don't believe there is an OTLP exporter for HTTP, as I believe gRPC is the preferred communication method

**Reply by jkwatson** [reply]
We haven't implemented an HTTP/json OTLP exporter. We do convert to OTLP-json in the otlp-logging exporter, so it would probably be a reasonably small amount of effort to implement it.

**Reply by wdengw** [reply]
Yes, OTLP with gRPC through the use of OtlpGrpcSpanExporter works.  The problem is, often,  organizations open up limited ports for many reasons.  HTTP and HTTPS are the common ports that organizations manage to open.  OTLP over HTTP is more suitable in those situation.  So, I am looking for such a SpanExporter.

**Reply by jkwatson** [reply]
Would you be interested in contributing the implementation?

**Reply by wdengw** [reply]
OpenTelemetry-Go has support for OTLP/HTTP. https://github.com/open-telemetry/opentelemetry-go/tree/main/exporters/otlp/otlphttp.  Is there plan to provide one on OpenTelemetry-Java?

**Reply by wdengw** [reply]
We are currently evaluating what options OpenTelemetry-Java provides.

**Reply by jkwatson** [reply]
There is no plan right now. If someone is interested in implementing it, it would be a welcome addition.

**Reply by anuraaga** [reply]
Agree that we can add an HTTP exporter based on okhttp. But just confirming, @wdengw, is the problem you have with gRPC ports? gRPC is based on HTTP or HTTPS so can use the same ports as any other HTTP server / client, even 80 / 443 if that's what you're using.

**Reply by idonava** [reply]
@wdengw, are you planning to implement the HTTP exporter? let me know if I can help in any way because I'm interested in that as well.

**Reply by wdengw** [reply]
@idonava  We don't have plan yet.
